### PR TITLE
Added hashDepositIntent method in ERC20GatewayBase contract

### LINKED ISSUE #121: Implement the DEPOSIT_INTENT_TYPEHASH constant and hashDepositIntent function in ERC20GatewayBase contract
- Create the contract `ERC20GatewayBase`
  - This contract will be inherited by the `ConsensusGatewayBase` contract
- Create a constant bytes32 variable `DEPOSIT_INTENT_TYPEHASH` for storing deposit intent type hash.
  - This will be used in `hashDepositIntent` function for the creation of `depositIntentHash_`
```solidity
bytes32 constant public DEPOSIT_INTENT_TYPEHASH = keccak256(
    abi.encode(
        "DepositIntent(uint256 amount,address beneficiary)"
    )  
);
```
- Create the function `hashDepositIntent`
  - Parameters:
    - `uint256 _amount`
    - `address _beneficiary`
  - Returns:
    - `bytes32 depositIntentHash_`
  - This function will hash the deposit intent
```solidity
function hashDepositIntent(
    uint256 _amount, 
    address _beneficiary
)
    public
    returns (bytes32 depositIntentHash_)     
```
- This hashing function `hashDepositIntent` will be used by both `ConsensusGateway` and `ConsensusCogateway` contracts

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains changes to add :
1. **hashDepositIntent** method in ERC20GatewayBase contract.
2. Unit test case

fixes #121

### CODE REVIEW

File: contracts/consensus-gateway/ERC20GatewayBase.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+contract ERC20GatewayBase {
+
+    /* Storage */
+
+    bytes32 constant public DEPOSIT_INTENT_TYPEHASH = keccak256(
+        "DepositIntent(uint256 amount,address beneficiary)"
+    );
+
```
Comment (deepesh-kn) [inline contracts/consensus-gateway/ERC20GatewayBase.sol:24]: Add a new line.

File: contracts/consensus-gateway/ERC20GatewayBase.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+contract ERC20GatewayBase {
+
+    /* Storage */
```
Comment (deepesh-kn) [inline contracts/consensus-gateway/ERC20GatewayBase.sol]:
```suggestion
    /* Constants */
```

File: contracts/consensus-gateway/ERC20GatewayBase.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
+pragma solidity >=0.5.0 <0.6.0;
+
```
Comment (deepesh-kn) [inline contracts/consensus-gateway/ERC20GatewayBase.sol:2]: Copyright is missing.

**Review by deepesh-kn — CHANGES_REQUESTED** [review]
Looks good to me. 
Just minor cosmetic changes.

File: contracts/consensus-gateway/ERC20GatewayBase.sol (hunk 1 above)
Comment (gulshanvasnani) [reply contracts/consensus-gateway/ERC20GatewayBase.sol:24]: Done

File: contracts/consensus-gateway/ERC20GatewayBase.sol (hunk 3 above)
Comment (gulshanvasnani) [reply contracts/consensus-gateway/ERC20GatewayBase.sol:2]: Done

File: contracts/consensus-gateway/ERC20GatewayBase.sol (hunk 2 above)
Comment (gulshanvasnani) [reply contracts/consensus-gateway/ERC20GatewayBase.sol]: Done

**Review by deepesh-kn — APPROVED** [review]
LGTM 👍

**Pull request merged**